File: unitri-ppi/aulas-spring/src/main/java/com/example/aulasspring/controller/VeiculoController.java

package com.example.aulasspring.controller;


import com.example.aulasspring.model.Veiculo;
import com.example.aulasspring.service.CategoriaService;
import com.example.aulasspring.service.VeiculoService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Controller;
import org.springframework.validation.BindingResult;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.servlet.ModelAndView;

import javax.validation.Valid;

@Controller
@RequestMapping(value = "/veiculos")
public class VeiculoController {

    @Autowired
    private VeiculoService service;

    @Autowired
    private CategoriaService categoriaService;

    @RequestMapping(method = RequestMethod.GET)
    public ModelAndView findAll() {

        ModelAndView mv = new ModelAndView("/veiculos");
        mv.addObject("veiculos", service.findAll());

        return mv;
    }

    @RequestMapping(value = "/add", method = RequestMethod.GET)
    public ModelAndView add(Veiculo veiculo) {

        ModelAndView mv = new ModelAndView("/veiculo");
        mv.addObject("veiculo", veiculo);
        mv.addObject("categorias", categoriaService.findAll());

        return mv;
    }

    @RequestMapping(value = "/edit/{id}", method = RequestMethod.GET)
    public ModelAndView edit(@PathVariable("id") Long id) {

        return add(service.findOne(id));
    }

    @RequestMapping(value = "/delete/{id}", method = RequestMethod.GET)
    public ModelAndView delete(@PathVariable("id") Long id) {

        service.delete(id);
        return findAll();
    }

    @RequestMapping(value = "/save", method = RequestMethod.POST)
    public ModelAndView save(@Valid Veiculo veiculo, BindingResult result) {

        if (result.hasErrors()) {
            return add(veiculo);
        }

        if (veiculo.getCategoria() != null) {
            veiculo.setCategoria(
                    categoriaService.findOne(veiculo.getCategoria().getId())
            );
        }
        service.save(veiculo);

        return findAll();
    }
}

File: unitri-ppi/aula-07/src/main/kotlin/com/example/aula07/controller/VeiculoController.kt
package com.example.aula07.controller

import com.example.aula07.model.Veiculo
import com.example.aula07.service.VeiculoService
import org.springframework.stereotype.Controller
import org.springframework.validation.BindingResult
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.servlet.ModelAndView
import javax.validation.Valid


@Controller
@RequestMapping("/veiculos")
class VeiculoController(
    val service: VeiculoService
) {

    @GetMapping
    fun findAll(): ModelAndView =
        ModelAndView("/veiculos").addObject("veiculos", service.findAll())

    @GetMapping("/add")
    fun add(veiculo: Veiculo): ModelAndView = ModelAndView("/veiculo").addObject("veiculo", veiculo)

    @GetMapping("/edit/{id}")
    fun edit(@PathVariable("id") id: Long): ModelAndView = add(service.findOne(id))

    @GetMapping("/delete/{id}")
    fun delete(@PathVariable("id") id: Long): ModelAndView {
        service.delete(id)
        return findAll()
    }

    @PostMapping("/save")
    fun save(@Valid veiculo: Veiculo, result: BindingResult): ModelAndView {

        if (result.hasErrors()) {
            return add(veiculo)
        }
        service.save(veiculo)
        return findAll()
    }

}